packages/next/src/server/lib/incremental-cache/fetch-cache.ts

```diff
@@ -233,19 +253,20 @@ export default class FetchCache implements CacheHandler {
           throw new Error(`invalid response from cache ${res.status}`)
         }
 
-        const cached = await res.json()
+        const cached: IncrementalCacheValue = await res.json()
+        const cachedParsed = zCachedFetchValue.safeParse(cached)
 
-        if (!cached || cached.kind !== 'FETCH') {
+        if (!cached || !cachedParsed.success) {
```

Comment:
If you're parsing and validating the data, use the validated structure here to ensure no extra fields are added by mistake:

```ts
const json = await res.json()
const parsed = zCachedFetchValue.safeParse(json)

if (!parsed.success) {
  this.debug && console.log({ json })
  throw new Error('invalid cache value')
}

const { data: cached } = parsed
```

Reply:
`data` on parsed, so that last line won't work <picture data-single-emoji=":thinkies:" title=":thinkies:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/thinkies/d164ba479fed9c0d.png" alt=":thinkies:" align="absmiddle"></picture>